.github/workflows/build.yml

```diff
@@ -63,3 +63,24 @@ jobs:
 
       - name: Upload to codecov.io
         uses: codecov/codecov-action@v1.0.6
+
+  no-default-features:
```

Comment:
I would suggest adding an actual compile with a `no-std` target, otherwise it is easy to miss things

Reply:
I was thinking about using https://github.com/hobofan/cargo-nono. do you mean using that `cargo build --target thumbv6m-none-eabi` "hack"?

Reply:
I'll do that in a separate PR, dealing with CI is often a lot of noise and I'd really like to get this PR merged :)

Reply:
this works pretty well: https://github.com/async-rs/async-std/blob/master/.github/workflows/ci.yml#L83-L99